src/Styleguide/Elements/Responsive.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react"
+import createContext, { ConsumerProps } from "create-react-context"
+
+// FIXME: Replace with React.createContext after React 16.3+ upgrade
+const ResponsiveContext = createContext({})
+
+export interface Breakpoints {
+  [breakpoint: string]: string
+}
+
+export interface ResponsiveProviderProps {
+  initialBreakpoint?: string
+  breakpoints: Breakpoints
+}
+
+export interface BreakpointState {
+  [breakpoint: string]: boolean
+}
+
+export interface ResponsiveProviderState {
+  breakpoints: BreakpointState
+  mediaMatchers: MediaQueryList[]
+  mqHandler: MediaQueryListListener | null
```

Comment:
I know that `mq` is an often used abbreviation for `media-query` but all these abbreviations being introduced in the context of breakpoints aren’t helping readability imo :/ Can we use actual words or is that an uphill battle with regards to 3rd party dependencies?

Reply:
Yeah, we can. This component has no 3rd party dependencies so we can update the language to whatever we want it to be.

Reply:
@alloy the old version piggy backed on https://github.com/damassi/react-responsive-decorator but this version skips all of that.

Reply:
@zephraph which make me think, once we have some time can you make a note to replace the old `<Responsive>` component where its used, or add a note deprecating it at the call sites?  I believe there's only a few places in Publishing but should be a pretty seamless transition.

Reply:
I can certainly do that!